esi/lib/node.py

```diff
@@ -16,93 +17,125 @@
 
 
 def network_list(connection, filter_node=None, filter_network=None):
-    """List nodes and their network attributes
-
-    :param connection: An OpenStack connection
-    :type connection: :class:`~openstack.connection.Connection`
-    :param node: Name or UUID of the node
-    :param network: The name or ID of a network
-    """
 
-    if filter_node:
-        machine = connection.baremetal.find_node(filter_node, ignore_missing=False)
-        ports = connection.list_nics_for_machine(uuid=machine.id)
-
-        node_name = machine.name
-    else:
-        ports = None
-        nodes = None
+    def get_baremetal_info(filter_node):
+        baremetal_ports = None
+        baremetal_nodes = None
+
+        if filter_node:
+            baremetal_nodes = [connection.baremetal.find_node(name_or_id=filter_node,
+                                                              ignore_missing=False)]
+            baremetal_ports = [connection.list_nics_for_machine(uuid=baremetal_nodes[0].id)]
```

Comment:
It may be better to use `connection.baremetal.ports` here. It's just more consistent with how we call the sdk; and for someone experienced with OpenStack, it's easier to understand what it does. In addition, if we ever want to fine tune the call to get the ports, the underlying command gives us that option, while `list_nics_for_machine` does not.